docs/utreexo.md

```diff
@@ -0,0 +1,72 @@
+
+<p align="center" width="100%">
+  <img src="./img/utreexo_pollard.svg" alt=""/>
+</p>
+
+Although the new scheme is more memory consuming it has several advantages:
+- Better UTXO locality: the leaves that are spent soon are clustered at the right, so the proof size is optimal in most cases (see some [cool charts](https://github.com/mit-dci/utreexo/issues/257)); at the same time the proof size for old UTXOs is also amortized thanks to the "hollowing";
+- The removal algorithm does not mess much with the tree structure and requires less updates of the intermediate nodes.
+
+### Node positions
+
+Another important implementation detail that will help with understanding is how nodes positions are calculated. In short, the current pollard is extended to the nearest perfect binary tree (i.e. the bottom most row length is power of two) and then we go row by row bottom up, enumerating both the existing and virtual nodes.
+
+<p align="center" width="100%">
+  <img src="./img/utreexo_positions.svg" alt=""/>
+</p>
+
+This allows more convenient navigation through the pollard and some bit tricks.
+
+## Stump
+
+Stump is the "light" version of Utreexo accumulator that contains the minimal necessary data required to verify inclusion proofs and add/remove leaves. It is not sufficient to generate proofs though — for that you need the full pollard, which is quite large.  
+Stump is a perfect fit for a ZK client where maintaining large state is prohibitively expensive.
+
+More specifically, stump consists of roots and the number of leaves (as we know this is the total number of UTXOs). Note that there are differences in how roots are stored and represented compared to the vanilla implementation:
+- Roots are sorted top down
+- There are no placeholders for rows that do not have roots
+- Empty roots are legit, it just means its children (aka nieces) are empty too
+
+<p align="center" width="100%">
+  <img src="./img/utreexo_stump.svg" alt=""/>
+</p>
+
+Note that previously we had either a root or "null" for every row in the forest, but in stump we have to use a different approach to determine to which row a particular root belongs. A simple rule is if the row length is even we skip it, otherwise (odd) the root is added. Another way to see this is to look at the binary representation of the leaves number and for each "1" there should be a root in the according row.
+
+Also it's worth noting that before "null" was indicating a missing roots whereas in stump empty root just means its children are empty too.
+
+## Batch proof
+
+Batch inclusion proofs were introduced in the Utreexo paper (see Appendix) and they can reduce the total size of proof(s) significantly. But the batch delete algorithm was pretty complex and IO intensive. The pollard scheme streamlined batch deletions and made batch proofs more practical.
+
+<p align="center" width="100%">
+  <img src="./img/utreexo_proof.svg" alt=""/>
```

Comment:
It is not clear what is being proven here.

Reply:
Inclusion :) I've added a sentence what batch proofs are consist of